components/oauth-client/src/a2.rs

```diff
@@ -73,23 +78,24 @@ impl A2 {
 impl OAuth2Provider for A2 {
     fn authenticate(&self,
                     config: &OAuth2Cfg,
-                    client: &Client,
+                    client: &HttpClient,
                     code: &str)
                     -> Result<(String, OAuth2User)> {
         let url = config.token_url.to_string();
-        let params = format!("client_id={}&client_secret={}&grant_type=authorization_code&\
-                              code={}&redirect_uri={}",
-                             config.client_id, config.client_secret, code, config.redirect_url);
-
-        let mut headers = Headers::new();
-        headers.set(Accept(vec![qitem(mime::APPLICATION_JSON)]));
-        headers.set(ContentType::form_url_encoded());
-
-        let mut resp = client.post(&url)
-                             .headers(headers)
-                             .body(params)
-                             .send()
-                             .map_err(Error::HttpClient)?;
+        let body = format!("client_id={}&client_secret={}&grant_type=authorization_code&code={}&\
+                            redirect_uri={}",
+                           config.client_id, config.client_secret, code, config.redirect_url);
```

Comment:
Leveraging [`format!`'s named parameters](http://doc.rust-lang.org/1.34.1/std/fmt/index.html#named-parameters) makes it easier to ensure the correct mapping and avoid introducing errors later if fields are added or removed:
```rust
        let body = format!("client_id={client_id}&client_secret={client_secret}&\
                            grant_type=authorization_code&code={code}&redirect_uri={redirect_uri}",
                           client_id = config.client_id,
                           client_secret = config.client_secret,
                           code = code,
                           redirect_uri = config.redirect_url);
```

Reply:
Generally I prefer the positional formatting, but this may be useful in some cases like oauth.  Will consider for a separate PR.